### PR TITLE
Fix deleting tag issue from nodes in RAC-6680

### PR DESCRIPTION
**Pull request by lanchongyizu**
* FIX the issue that Cannot delete a tag from a node if tag contains spaces
(https://rackhd.atlassian.net/browse/RAC-6680)
* throw NotFoundError when tagName not found in nodes for API `/nodes/{identifier}/tags/{tagName}
`

depends on: https://github.com/RackHD/RackHD/pull/923

@RackHD/corecommitters @pengz1

### CODE REVIEW

File: lib/api/2.0/tags.js [hunk 1]
```diff
@@ -10,18 +10,6 @@ var _ = injector.get('_'); // jshint ignore:line
 var Errors = injector.get('Errors');
 var parseUrl = require('url').parse;
 
-function getTagName(req) {
-    //TODO: sway 1.0.0 implements a fix for this, but it is lacking in the ^0.6.0 versions
-    //      so we duplicate the 1.0.0 implementation here.
-    // Ref: https://github.com/apigee-127/sway/commit/1b7acd9743aae2f2f94057d3f8911e2af72614f5
-    var pathObject = req.swagger.operation.pathObject;
-    var pathMatch = pathObject.regexp.exec(parseUrl(req.url).pathname);
-    var value = decodeURIComponent(pathMatch[_.findIndex(pathObject.regexp.keys, function(key) {
```
Comment (nortonluo) [inline lib/api/2.0/tags.js]: Why we drop the tag URL decode phase here? To upgrade swagger-express-mw from 0.6 to 0.7 is too risky which may involve other compatibility issue. Did we test the APIs which also depends on this lib?
Please refer to the comments:
https://github.com/RackHD/on-http/pull/272/files

Comment (lanchongyizu) [reply lib/api/2.0/tags.js]: @nortonluo The comments in https://github.com/RackHD/on-http/pull/272/files support our upgrading to 0.7. So I don't understand why you mentioned the comments in https://github.com/RackHD/on-http/pull/272/files.

**Comment by lanchongyizu** [comment]
The concourse-ci fails because of throwing NotFoundError when tagName not found in nodes for API `/nodes/{identifier}/tags/{tagName}`. Our FIT accepts 204 rather than 404 when tagName not found in nodes, so we need to update FIT.

**Comment by lanchongyizu** [comment]
```
Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:344:11)
    at /RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/lib/connect_middleware.js:83:26
    at /RackHD/on-http/node_modules/lodash/index.js:3073:15
    at baseForOwn (/RackHD/on-http/node_modules/lodash/index.js:2046:14)
    at /RackHD/on-http/node_modules/lodash/index.js:3043:18
    at Function.<anonymous> (/RackHD/on-http/node_modules/lodash/index.js:3346:13)
    at finishConnect (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/lib/connect_middleware.js:82:17)
    at bound (domain.js:287:14)
    at runBound (domain.js:300:12)
    at null.<anonymous> (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipewor
ks.js:72:17)
    at null.<anonymous> (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipewor
ks.js:179:12)
    at bound (domain.js:287:14)
    at runBound (domain.js:300:12)
    at null.<anonymous> (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipewor
ks.js:72:17)
    at postFlight (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/lib/bagpipes.js:220:3)
    at /RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/lib/bagpipes.js:174:9
    at swagger_raw (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/fittings/swagger_raw.js:43:7)
    at null.<anonymous> (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/lib/bagpipes.js:171:7)
    at bound (domain.js:287:14)
    at runBound (domain.js:300:12)
    at null.pipeline (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipeworks.
js:72:17)
    at Runner.flow (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipeworks.js
:223:19)
at Pipeworks.flow (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipeworks
.js:135:17)
    at Pipeworks.siphon (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipewor
ks.js:186:19)
    at null.<anonymous> (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/lib/bagpipes.js:59:41)
    at bound (domain.js:287:14)
    at runBound (domain.js:300:12)
    at null.pipeline (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipeworks.
js:72:17)
    at Runner.flow (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipeworks.js
:223:19)
    at Pipeworks.flow (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/node_modules/pipeworks/pipeworks
.js:135:17)
    at Bagpipes.play (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/node_modules/bagpipes/lib/bagpipes.js:61:6)
    at /RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/lib/connect_middleware.js:110:25
    at Runner.applyMetadata (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/index.js:112:5)
    at middleware (/RackHD/on-http/node_modules/swagger-express-mw/node_modules/swagger-node-runner/lib/connect_middleware.js:45:14)
    at Layer.handle [as handle_request] (/RackHD/on-http/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/RackHD/on-http/node_modules/express/lib/router/index.js:317:13)
    at /RackHD/on-http/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/RackHD/on-http/node_modules/express/lib/router/index.js:335:12)
    at next (/RackHD/on-http/node_modules/express/lib/router/index.js:275:10)
    at swaggerUI (/RackHD/on-http/node_modules/swagger-tools/middleware/swagger-ui.js:167:12)

```

**Pull request merged**